src/noit_clustering.c

```diff
@@ -898,6 +901,104 @@ cluster_topo_cb(void *closure,
+    return MTEV_HOOK_CONTINUE;
+  }
+  // If the batch data is the default and not explicitly provided in the config, don't write it out
+  if (!batch_size_from_config) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  char config_str[1024];
+  snprintf(config_str, sizeof(config_str), "%"PRIu32, batch_size);
+  char *batch_conf = (char *)xmlGetProp(node, (xmlChar *)"batch_size");
+  if (batch_conf) {
+    xmlFree(batch_conf);
+    return MTEV_HOOK_ABORT;
+  }
+  xmlSetProp(node, (xmlChar *)"batch_size", (xmlChar *)config_str);
+  return MTEV_HOOK_CONTINUE;
+}
+
+static mtev_hook_return_t
+reconnoiter_specific_write_node_config_xml_cb(void *closure, mtev_cluster_t *cluster, uuid_t node_id, xmlNodePtr node) {
+  if(strcmp(mtev_cluster_get_name(cluster), NOIT_MTEV_CLUSTER_NAME)) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  // No node specific noit configs needed
+  return MTEV_HOOK_CONTINUE;
+}
+
+static mtev_hook_return_t
+reconnoiter_specific_write_cluster_config_json_cb(void *closure, mtev_cluster_t *cluster, struct json_object *obj) {
+  if(strcmp(mtev_cluster_get_name(cluster), NOIT_MTEV_CLUSTER_NAME)) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  // If the batch data is the default and not explicitly provided in the config, don't write it out
+  if (!batch_size_from_config) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  MJ_KV(obj, "batch_size", MJ_UINT64(batch_size));
+  return MTEV_HOOK_CONTINUE;
+}
+
+static mtev_hook_return_t
+reconnoiter_specific_write_node_config_json_cb(void *closure, mtev_cluster_t *cluster, uuid_t node_id, struct json_object *obj) {
+  if(strcmp(mtev_cluster_get_name(cluster), NOIT_MTEV_CLUSTER_NAME)) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  // No node specific noit configs needed
+  return MTEV_HOOK_CONTINUE;
+}
+
+static mtev_hook_return_t
+reconnoiter_specific_read_cluster_config_cb(void *closure, mtev_cluster_t *cluster, mtev_conf_section_t *conf) {
+  if(strcmp(mtev_cluster_get_name(cluster), NOIT_MTEV_CLUSTER_NAME)) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  if (!conf) {
+    return MTEV_HOOK_CONTINUE;
+  }
+
+  char bufstr[1024];
+  char *endptr = NULL;
+  if(mtev_conf_get_stringbuf(*conf, "@batch_size", bufstr, sizeof(bufstr))) {
+    uint32_t local_batch_size = strtoll(bufstr, &endptr, 10);
+    if(*endptr || local_batch_size == 0) {
+      mtevL(mtev_error, "Invalid batch size (%s) provided.... keeping previous batch size (%" PRIu32 ")\n", bufstr, batch_size);
+    }
+    else
+    {
+      batch_size = local_batch_size;
+      batch_size_from_config = true;
+    }
+  }
+
+  return MTEV_HOOK_CONTINUE;
+}
+
+static mtev_hook_return_t
+reconnoiter_specific_read_node_config_cb(void *closure, mtev_cluster_t *cluster, uuid_t node_id, mtev_conf_section_t *conf) {
+  if(strcmp(mtev_cluster_get_name(cluster), NOIT_MTEV_CLUSTER_NAME)) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  if (!conf) {
+    return MTEV_HOOK_CONTINUE;
+  }
+  // No node specific noit configs needed
+  return MTEV_HOOK_CONTINUE;
```

Comment:
If it's always CONTINUE why are we doing these other things?

Reply:
Documentation purposes. If we _did_ put any special code in there (where the comment is), we'd want to be able to continue or error out if there's a problem.

Technically, these node-specific hook callbacks are not necessary, as we are not using any node-specific stuff. I just have them there because I think it makes it clear where things would go later if we needed to add stuff; however, if you think it complicates/confuses things, I can take the functions completely out, as they do not need to be here.